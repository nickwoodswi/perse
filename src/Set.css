.set{
    display: flex;
    width: 100%;
    align-items: center;
    background-color: green;
    margin: 2px;
}
    .set-spec{
        width: 100%;
        padding: 5px;
    }
    .set-edit-buttons{
        display: flex;

    }
        .move-set-button{
            margin: 0;
            width: 70px;
        }