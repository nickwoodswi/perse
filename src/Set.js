import React, {Component} from 'react'
import './Set.css'
import AddSets from './AddSets'
import Weight from './Weight'
import Distance from './Distance'
import Reps from './Reps'
import Tempo from './Tempo'
import Rest from './Rest'
import Subrest from './Subrest'

class Set extends Component {
    render() {
        let exerciseName = []
        this.props.exerciseTypes.map(exerciseType => {
            if (exerciseType.exercise_types_id == this.props.set.exercise_types_id) {
                exerciseName.push(exerciseType.exercise_types_name)
            }
        })
        return(
            <div className="set">
                <div className="exercise-summary">
                    <div className="set-number">
                        <AddSets 
                            setId={this.props.id}
                            targetSet={this.props.set}
                            addSets={this.props.addSets} />
                    </div>
                    <div className="set-spec">
                        <b>{this.props.set.exercise_types_name}</b> <Reps reptype={this.props.set.rep_type} reps={this.props.set.reps} />
                        <Weight weight={this.props.set.weight} />
                        <Distance distance={this.props.set.sub_distance} />
                        <Tempo speed={this.props.set.tempo_time} />
                        <Subrest time={this.props.set.subrest_time} />
                        <Rest rest={this.props.set.rest_time} />
                    </div>
                </div>
                <div className="set-edit-buttons">
                    <button 
                        onClick={() => this.props.moveSet(this.props.id, -1)}
                        className="move-set-button">
                            UP</button>
                    <button 
                        onClick={() => this.props.moveSet(this.props.id, 1)}
                        className="move-set-button">
                            DOWN</button>
                    <button 
                        onClick={() => this.props.deleteSet(this.props.id)} 
                        className="delete-set-button">
                            DELETE</button>
                </div>
            </div>
        )
    }
}

export default Set