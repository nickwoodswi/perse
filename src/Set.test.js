import React from 'react';
import { render } from '@testing-library/react';
import Set from './Set';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders set component', () => {
    render(<MemoryRouter><Set /></MemoryRouter>);
});

it('Set component renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><Set /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});