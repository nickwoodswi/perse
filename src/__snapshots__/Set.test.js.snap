// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set component renders the UI as expected 1`] = `
<div
  className="set"
>
  <div
    className="exercise-summary"
  >
    <div
      className="set-number"
    >
      <input
        className="set-number-input"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="set-spec"
    >
      <b />
       
       No rep type selected
    </div>
  </div>
  <div
    className="set-edit-buttons"
  >
    <button
      className="move-set-button"
      onClick={[Function]}
    >
      UP
    </button>
    <button
      className="move-set-button"
      onClick={[Function]}
    >
      DOWN
    </button>
    <button
      className="delete-set-button"
      onClick={[Function]}
    >
      DELETE
    </button>
  </div>
</div>
`;
